shells/pipes-shell/source/verbs/run-arc.js

```diff
@@ -117,3 +117,9 @@ const updateParticleInPlan = (plan, particleId, particleName, providedSlotId) =>
 export const stopArc = async ({arcId}, runtime) => {
   runtime.stop(arcId);
 };
+
+export const uiEvent = async({particleId, eventlet}, runtime) => {
+  const arc = [...runtime.arcById.values()].find(
+      arc => arc.activeRecipe.particles.find(p => p.id.toString() === particleId));
```

Comment:
This is pretty hard to read. I also feel like this should be a method on either Arc or Recipe: getParticle(particleId: Id). If you don't mind, could you add something like that (with a test)? Adding a TODO and a follow up PR would be fine too.

It'd be great to get into the mindset of being on the lookout for missing functionality in our public Arc/Runtime API, and adding that functionality properly, rather than hacking around it in shell code like we've been doing so far.

Reply:
done - added methods to runtime and to recipe.